src/CMakeLists.txt

```diff
@@ -65,7 +65,8 @@ set(HEADER_FILES
   ${LIBCELLML_CONFIG_H}
 )
 
-add_compiler_export_flags()
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
```

Comment:
These values should be set as properties on the target not through manipulating the default CMake variables.